app/src/androidTest/java/com/github/se/project/ui/profile/EditTutorProfileTest.kt

```diff
@@ -0,0 +1,172 @@
+        Profile(
+            uid = "12345",
+            googleUid = "67890",
+            firstName = "John",
+            lastName = "Doe",
+            phoneNumber = "1234567890",
+            role = Role.TUTOR,
+            section = Section.IN,
+            academicLevel = AcademicLevel.MA2,
+            languages = listOf(Language.ENGLISH),
+            subjects = listOf(Subject.ALGEBRA),
+            schedule = List(7) { List(12) { 0 } })
+    // Set the screen in the test environment
+    composeTestRule.setContent { EditTutorProfile(mockNavigationActions, mockViewModel) }
+
+    composeTestRule.onNodeWithTag("editTutorProfileAcademicLevelDropdown").performClick()
+    composeTestRule
+        .onNodeWithTag("editTutorProfileAcademicLevelDropdownItem-MA4")
+        .assertIsDisplayed()
+        .performClick()
+
+    composeTestRule.onNodeWithTag("editTutorProfileSectionDropdown").performClick()
+    composeTestRule
+        .onNodeWithTag("editTutorProfileSectionDropdownItem-GM")
+        .assertIsDisplayed()
+        .performClick()
+
+    composeTestRule.onNodeWithTag("editTutorProfileConfirmButton").performClick()
+
+    assertEquals(AcademicLevel.MA4, mockViewModel.currentProfile.value?.academicLevel)
+    assertEquals(Section.GM, mockViewModel.currentProfile.value?.section)
+  }
+
+  @Test
+  fun editTutorClosePage() {
+    (mockViewModel.currentProfile as MutableStateFlow).value =
+        Profile(
+            uid = "12345",
+            googleUid = "67890",
+            firstName = "John",
+            lastName = "Doe",
+            phoneNumber = "1234567890",
+            role = Role.TUTOR,
+            section = Section.IN,
+            academicLevel = AcademicLevel.MA2,
+            languages = listOf(Language.ENGLISH),
+            subjects = listOf(Subject.ALGEBRA),
+            schedule = List(7) { List(12) { 0 } })
+    // Set the screen in the test environment
+    composeTestRule.setContent { EditTutorProfile(mockNavigationActions, mockViewModel) }
+
+    composeTestRule.onNodeWithTag("editTutorProfileCloseButton").performClick()
+    Mockito.verify(mockNavigationActions).goBack()
+  }
+
+  @Test
+  fun editTutorProfileScreen_ToastMessage() {
+    (mockViewModel.currentProfile as MutableStateFlow).value =
+        Profile(
+            uid = "12345",
+            googleUid = "67890",
+            firstName = "John",
+            lastName = "Doe",
+            phoneNumber = "1234567890",
+            role = Role.TUTOR,
+            section = Section.IN,
+            academicLevel = AcademicLevel.MA2,
+            schedule = List(7) { List(12) { 0 } })
+    // Set the screen in the test environment
+    composeTestRule.setContent {
+      EditTutorProfile(navigationActions = mockNavigationActions, mockViewModel)
+    }
+    composeTestRule.onNodeWithTag("editTutorProfileConfirmButton").performClick()
+
+    Mockito.verify(mockNavigationActions, Mockito.never()).navigateTo(Mockito.anyString())
```

Comment:
Same as above, could be interesting to test other cases. As is this works though.